src/js/base/ripe.js

```diff
@@ -35,6 +35,10 @@ ripe.Ripe.prototype.init = function(brand, model, options) {
     var hasParts = this.parts && Object.keys(this.parts).length !== 0;
 
     this.ready = hasParts;
+
+    this.bind("pre_parts", function() {

```

Comment:
why not post_parts ?

Reply:
The idea here is that partsHistory would contain all the previous states, so the `pre_parts` event is used to store the current configuration before a change. The `post_parts` event contains the parts after the changes.

Reply:
ok so the stack does not contain the current state right ?

Reply:
Right, only the previous states.

Reply:
yeahh, and I know that the current implementation. But I think we should improve it and have a sliding strategy (stack pointer) so that we can have undo and redo... That would be cool